pkg/storaged/lvol-tabs.jsx

```diff
@@ -141,8 +231,68 @@ var BlockVolTab = React.createClass({
             lvol_rename(lvol);
         }
 
-        function resize() {
-            lvol_resize(self.props.client, lvol);
+        function get_info(block) {
+            if (block) {
+                if (block.IdUsage == 'crypto' && client.blocks_crypto[block.path]) {
+                    var encrypted = client.blocks_crypto[block.path];
+                    var cleartext = client.blocks_cleartext[block.path];
+
+                    if (!encrypted.Resize) {
+                        info = { };
+                        shrink_excuse = grow_excuse = _("Encrypted volumes can not be resized here.");
+                    } else if (!cleartext) {
+                        info = { };
+                        shrink_excuse = grow_excuse = _("Encrypted volumes need to be unlocked before they can be resized.");
+                    } else {
+                        return get_info(cleartext);
+                    }
+                } else if (block.IdUsage == 'filesystem' && client.fsys_info[block.IdType]) {
+                    info = client.fsys_info[block.IdType];
+                    if (!info.can_shrink)
+                        shrink_excuse = cockpit.format(_("$0 filesystems can not be made smaller."),
+                                                       block.IdType);
+                    if (!info.can_grow)
+                        grow_excuse = cockpit.format(_("$0 filesystems can not be made larger."),
+                                                     block.IdType);
+                } else if (block.IdUsage == 'raid') {
+                    info = { };
+                    shrink_excuse = grow_excuse = _("Physical volumes can not be resized here.");
+                } else if (client.vdo_overlay.find_by_backing_block (block)) {
+                    info = {
+                        can_shrink: false,
+                        can_grow: true,
+                        grow_needs_unmount: false
```

Comment:
Add  `shrink_needs_unmount` as well, or drop the fields which are `false`, for symmetry?

Reply:
The logic here is that `shrink_needs_unmount` only needs to be specified when `can_shrink` is true.  Otherwise it doesn't apply.